webapp/src/components/modals/attach_comment_to_issue/attach_comment_to_issue.tsx

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import {Modal} from 'react-bootstrap';
+import {Theme} from 'mattermost-redux/types/preferences';
+import {Post} from 'mattermost-redux/types/posts';
+
+import FormButton from '../../form_button';
+import Input from '../../input';
+import Validator from '../../validator';
+
+import GitlabIssueSelector from '../../gitlab_issue_selector';
+import {getErrorMessage} from '../../../utils/user_utils';
+
+const initialState = {
+    submitting: false,
+    issueValue: null,
+    error: '',
+};
+
+interface PropTypes {  
+    post: Post,
+    theme: Theme,
+    visible: boolean,
+    actions: {
+        close: () => {
+            type: string;
+        };
+        create: (payload: CommentBody) => Promise<{
+            error?: ErrorType;
+        }>;
+    }
+}
+
+interface StateTypes {
+    submitting: boolean,
+    issueValue: Issue | null,
+    error: string,
+}
+
+export default class AttachCommentToIssueModal extends PureComponent<PropTypes, StateTypes> {
+    validator: Validator;
+
+    constructor(props: PropTypes) {
+        super(props);
+        this.state = initialState;
+        this.validator = new Validator();
+    }
+
+    handleCreate = async (e: React.FormEvent<HTMLFormElement> | Event) => {
+        e.preventDefault();
+
+        if (!this.validator.validate()) {
+            return;
+        }
+
+        const comment = {
+            project_id: this.state.issueValue?.project_id,
+            iid: this.state.issueValue?.iid,
+            comment: this.props.post.message,
+            post_id: this.props.post.id,
+            web_url: this.state.issueValue?.web_url,
+        };
+
+        this.setState({submitting: true});
+
+        const created = await this.props.actions.create(comment);
+        if (created.error) {
+            const errMessage = getErrorMessage(created.error.message);
+            this.setState({error: errMessage, submitting: false});
+            return;
+        }
+
+        this.handleClose();
+    };
+
+    handleClose = () => {
```

Comment:
Try to use `useCallback` or `useMemo` in every React component for optimization.